pkg/admission/webhook.go

```diff
@@ -0,0 +1,112 @@
+package admission
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	herrors "github.com/horizoncd/horizon/core/errors"
+	"github.com/horizoncd/horizon/pkg/admission/models"
+	perror "github.com/horizoncd/horizon/pkg/errors"
+	"github.com/horizoncd/horizon/pkg/util/log"
+)
+
+const DefaultTimeout = 5 * time.Second
+
+var (
+	validatingWebhooks []Webhook
+)
+
+func Register(kind models.Kind, webhook Webhook) {
+	switch kind {
+	case models.KindValidating:
+		validatingWebhooks = append(validatingWebhooks, webhook)
+	}
+}
+
+func Clear() {
+	validatingWebhooks = nil
+}
+
+type validateResult struct {
+	err  error
+	resp *Response
+}
+
+type Request struct {
+	Operation    models.Operation       `json:"operation"`
+	Resource     string                 `json:"resource"`
+	ResourceName string                 `json:"resourceName"`
```

Comment:
Confused by the two fields: Resource and ResourceName. Could we choose any other names?

Reply:
Refer to RequestInfo struct, rename ResourceName to Name.